src/modules/vtol_att_control/vtol_att_control_main.cpp

```diff
@@ -666,6 +671,18 @@ void VtolAttitudeControl::calc_tot_airspeed() {
 	_airspeed_tot = _params.arsp_lp_gain * (_airspeed_tot - airspeed_raw) + airspeed_raw;
 }
 
+void VtolAttitudeControl::limit_min_thrust() {
+	// filter thrust command
+	_thrust_filt = _params.arsp_lp_gain*(_thrust_filt - _actuators_mc_in.control[3]) + _actuators_mc_in.control[3];
+	// prevent too low airspeeds over control surfaces in altitude control mode
+	if(_airspeed_tot < _params.mc_airspeed_min && _v_control_mode.flag_control_altitude_enabled && _thrust_low_bound < 0.001f) {
+		_thrust_low_bound = _thrust_filt;
+	}
+	if(_actuators_mc_in.control[3] > _thrust_low_bound + 0.05f || !_v_control_mode.flag_control_altitude_enabled) {
+		_thrust_low_bound = 0.0f;
+	}
```

Comment:
I don't understand what exactly you are trying to solve with _thrust_low_bound.
You're basically setting _thrust_low_bound *once* when airspeed is too low and you reset it to 0 when you throttle up in non-altitude mode?